win32ss/user/winsrv/consrv/include/conio.h

```diff
@@ -372,4 +372,9 @@ NTSTATUS ConioResizeBuffer(PCONSOLE /*PCONSRV_CONSOLE*/ Console,
 /* wcwidth.c */
 int mk_wcwidth_cjk(wchar_t ucs);
 
+// NOTE: The check against 0x80 is to avoid calling the helper function
+// for characters that we already know are not full-width.
+#define IS_FULL_WIDTH(wch)  \
+    ((wch) >= 0x80 && mk_wcwidth_cjk(wch) == 2)
```

Comment:
This is wrong if `wch` is a signed type

Reply:
it's WCHAR / wchar_t

Reply:
The signedness of wchar_t is implementation defined

Reply:
please let me know what you think about the second commit.